compiler/optimizer/LoopVersioner.cpp

```diff
@@ -2449,12 +2449,20 @@ bool opCodeIsHoistable(TR::Node *node, TR::Compilation *comp)
 bool TR_LoopVersioner::isExprInvariant(TR::Node *node, bool ignoreHeapificationStore)
    {
    _visitedNodes.empty();
-	   return isExprInvariantRecursive(node, ignoreHeapificationStore);
-	   }
+   return isExprInvariantRecursive(node, ignoreHeapificationStore);
+   }
 
-	bool TR_LoopVersioner::isExprInvariantRecursive(TR::Node *node, bool ignoreHeapificationStore)
+bool TR_LoopVersioner::isExprInvariantRecursive(TR::Node *node, bool ignoreHeapificationStore)
    {
    static const bool paranoid = feGetEnv("TR_paranoidVersioning") != NULL;
+
+   // Do not attempt to privatize BCD type nodes because doing so can result in creating direct loads of BCD types which are currently
+   // not handled correctly within Java. BCDCHK IL will attempt to recreate a call to the original Java API for the corresponding BCD
+   // IL and it needs to be able to materialize the node representing the original byte array object. This is not possible if the byte
+   // array is stored on the stack.
+   if (node->getType().isBCD())
+      return false;
```

Comment:
Unfortunately we'll need to guard this with `#ifdef J9_PROJECT_SPECIFIC` since the `isBCD` API is not available in OMR and this class is not yet extensible. Unfortunate, but hopefully a short lived reality as we look to eliminate BCD IL.

Reply:
Done.